### DISCUSSION TITLE
Trigger open_url_with_hints in a specific kitty window?

### DISCUSSION THREAD

**Discussion by 097115** [kovidgoyal/kitty#5698] [Q&A] [answered]
Is it possible somehow to trigger the `open_url_with_hints` action in a specific kitty window? Via the remote control? 

I tried mapping it to `shift+alt+z`, for instance, and then sending these keys via `send-text`, like this:

     kitty @ --to unix:/tmp/mykitty send-text \\x1b\\x5a

But it didn't work. (Wrong key sequence may be?)

**Comment by kovidgoyal** [comment]
open_url_with_hints is just sugar for running the hints kitten. You can run it manually with something like

```
map f1 launch --type=window --stdin-source=screen kitty +kitten hints
```

If you want to open it in a new window and send some key presses to it use combine with send-text but note that the kitten uses the kitty keyboard protocol, so you need to use the correct encoding (run kitty +kitten show_key -m kitty) to check that.

```
map f1 combine : launch ... : send_text kitty ...
```

see https://sw.kovidgoyal.net/kitty/faq/#how-do-i-map-key-presses-in-kitty-to-different-keys-in-the-terminal-program

**Comment by 097115** [comment]
@kovidgoyal 

Thank you so much for your answer, it certainly helped me understand things better. However, I still can't join all the puzzle pieces together, so may be I'd just lay out my initial problem, if you don't mind?

The point is, I'm running an IRC client, weechat, in one of the kitty windows (it always has the same `--title=IRC` just in case). And in weechat, there's an option to turn off all the buffer decorations and switch to the bare text via the `/window bare` command, mapped to `alt+l` (in this case links don't get split over several lines, and therefore always can be opened), 

So, I want to map these two consecutive key presses on my side, `alt+l` and `ctrl+shift+z`, to one binding. Or to call them somehow. An obvious solution would be something like 

    map f1 combine , send_text all \x1b\x6c , open_url_with_hints

But with this mapping, the second part, this `open_url_with_hints` happens actually **before** the `send_keys` part. I also tried to add ` , launch --type-background sh -c"sleep 1" , ` between those two but it didn't help.

Now, given your reply, there should be some smart way of joining all these pieces together, but I just can see it, sorry :(

So, at the moment, I ended with a mapping that simply runs a shell script in the background, and the said script is just something like `xdotool key Alt+l && sleep 0.5 && xdotool Ctrl+Shift+z`. Which seems to work, but, of course, if you could help me with some hints on cooking a native kitty-only solution, that would be way better and cleaner :)

Sorry for being that verbose, and also again for being some sort of blind and / or stupid. And thanks again for your help!

(BTW, we can't produce a mapping for a certain kitty window only, which would be defined by `--title`, for instance... Am I right?)

**Comment by kovidgoyal** [comment, marked as answer]
Your problem is there are two independent processes here, the terminal
emulator and weechat. kitty will send the key event and then immediately
run open_url_with_hints, you will get your desired result only if
weechat receives and reacts to that key press before kitty runs
open_url_with_hints. That is evidently not happening. 

The only robust way to solve this is to have weechat run a script on the
key press that uses remote control to trigger the hints kitten. However,
a simpler way is to just write a custom kitten that sends the key press
and then sleeps for a second (or whatever) and then runs open_url_hints.
See https://sw.kovidgoyal.net/kitty/kittens/custom/ 

Or run kitty from master and use the new sleep action I just created in
combine to avoid having to write your own kitten.

**Comment by 097115** [comment]
Thanks, Kovid, really appreciate your help!
